programs/test/benchmark.c

```diff
@@ -1001,11 +1001,18 @@ int main( int argc, char *argv[] )
             mbedtls_ecdh_setup( &ecdh_srv, curve_info->grp_id );
             mbedtls_ecdh_setup( &ecdh_cli, curve_info->grp_id );
 
+#if defined(MBEDTLS_ECDH_LEGACY_CONTEXT)
```

Comment:
Moving this guard here duplicate a lot of measurements: before this change we had two mutually exclusive but overlapping measurement sets, but now we perform them twice. The difference is that here we measure both sides of the handshake and in the previous set we had Curve448 too.

It feels like the program was better before this change. Am I missing something?

Reply:
These are different tests ("full handshake" vs old "handshake"), but we could of course disable the full handshakes if `MBEDTLS_ECDH_LEGACY_CONTEXT` is on.

Reply:
The way I see it, the application will rarely run both sides of the scheme and benchmarking one side is much more interesting. From this perspective the old handshake benchmarks make perfect sense. What is the advantage of benchmarking the full handshake?

Reply:
Our main application at the moment runs both sides and often locally (enclave to host/client), so I'm interested in the whole handshake performance. There's no advantage either way, it's a different test/benchmark, but feel free to remove it if it's not helpful.

Reply:
In that case, I am happy with leaving it as it is.